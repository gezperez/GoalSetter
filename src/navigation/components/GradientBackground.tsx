import React, { FC, memo } from 'react'
import { Dimensions, StyleSheet, ViewStyle } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'

import { Colors } from '~/utils'

const WIDTH = Dimensions.get('window').width

type GradientBackgroundProps = {
  height: number
}

const GradientBackground: FC<GradientBackgroundProps> = ({ height }) => (
  <LinearGradient
    start={{ x: 0, y: 0 }}
    end={{ x: 1, y: 0 }}
    colors={[Colors.HEADER_GRADIENT_START, Colors.HEADER_GRADIENT_END]}
    style={[styles.background, { height }]}
  />
)

type Styles = {
  background: ViewStyle
}

const styles = StyleSheet.create<Styles>({
  background: {
    width: WIDTH,
  },
})

export default memo(GradientBackground)
