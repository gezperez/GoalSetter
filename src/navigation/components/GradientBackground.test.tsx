import React from 'react'
import { render } from '@testing-library/react-native'

import GradientBackground from './GradientBackground'

describe('Background', () => {
  afterEach(jest.resetAllMocks)

  it('should render with required props', () => {
    const { toJSON } = render(<GradientBackground height={100} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
