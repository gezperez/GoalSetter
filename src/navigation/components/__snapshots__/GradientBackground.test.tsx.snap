// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background should render with required props 1`] = `
<BVLinearGradient
  colors={
    Array [
      4287440068,
      4283436913,
    ]
  }
  endPoint={
    Object {
      "x": 1,
      "y": 0,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Array [
      Object {
        "width": 750,
      },
      Object {
        "height": 100,
      },
    ]
  }
/>
`;
